ci-operator/step-registry/cucushift/installer/check/gcp/nested-virtualization/cucushift-installer-check-gcp-nested-virtualization-commands.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+if [ ! -f "${SHARED_DIR}/bastion_public_address" ] || [ ! -f "${SHARED_DIR}/bastion_ssh_user" ]; then
+    echo "ERROR: Failed to get bastion host info, abort. " && exit 1
+fi
+
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ssh_key_file_name="ssh-privatekey"
+ssh_key=${CLUSTER_PROFILE_DIR}/${ssh_key_file_name}
+bastion_dns=$(head -n 1 "${SHARED_DIR}/bastion_public_address")
+bastion_user=$(head -n 1 "${SHARED_DIR}/bastion_ssh_user")
+
+working_dir=$(mktemp -d)
+pushd "${working_dir}"
+
+ssh_proxy_cmd_template="ssh -i ${ssh_key} -o StrictHostKeyChecking=no -o ProxyCommand=\"ssh -i ${ssh_key} -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p ${bastion_user}@${bastion_dns}\" core@NODE_IP \"grep -cw vmx /proc/cpuinfo\" 1>stdout 2>stderr"
```

Comment:
here `stdout` is a file name?

Reply:
Yes, in the current directory. Any suggestion?

Reply:
```suggestion
ssh_proxy_cmd_template="ssh -i ${ssh_key} -o StrictHostKeyChecking=no -o ProxyCommand=\"ssh -i ${ssh_key} -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -W %h:%p ${bastion_user}@${bastion_dns}\" core@NODE_IP \"grep -cw vmx /proc/cpuinfo\" >stdout.log"
```
To avoid some reading confusion.

Reply:
Got it, thanks!